File: frontend/src/types/note.ts
export interface Notebook {
  id: number;
  user_id: number;
  name: string;
  description?: string;
  icon: string;
  is_default: boolean;
  note_count: number;
  created_at: string;
}

export interface NotebookCreate {
  name: string;
  description?: string;
  icon?: string;
}

export interface NotebookUpdate {
  name?: string;
  description?: string;
  icon?: string;
}

export interface Note {
  id: number;
  user_id: number;
  notebook_id?: number;
  learning_path_id?: number;
  title?: string;
  content: string;
  tags: string[];
  editor_mode: 'markdown' | 'rich_text';
  created_at: string;
  updated_at?: string;
}

export interface NoteCreate {
  notebook_id?: number;
  learning_path_id?: number;
  title?: string;
  content: string;
  tags?: string[];
  editor_mode?: 'markdown' | 'rich_text';
}

export interface NoteUpdate {
  notebook_id?: number;
  title?: string;
  content?: string;
  tags?: string[];
  editor_mode?: 'markdown' | 'rich_text';
}

File: backend/app/schemas/note.py
from pydantic import BaseModel
from datetime import datetime
from typing import List, Optional


class NoteCreate(BaseModel):
    """创建笔记"""
    notebook_id: int
    title: Optional[str] = None
    content: str
    tags: List[str] = []
    editor_mode: str = 'markdown'
    learning_path_id: Optional[int] = None


class NoteUpdate(BaseModel):
    """更新笔记"""
    title: Optional[str] = None
    content: Optional[str] = None
    tags: Optional[List[str]] = None


class NoteResponse(BaseModel):
    """笔记响应"""
    id: int
    user_id: int
    notebook_id: Optional[int] = None
    title: Optional[str] = None
    content: str
    tags: List[str]
    editor_mode: str = 'markdown'
    learning_path_id: Optional[int] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    
    class Config:
        from_attributes = True

